themes/default/content/blog/top-5-things-for-azure-devs-kubernetes-infrastructure/index.md

```diff
@@ -0,0 +1,566 @@
+			resourceGroupName := args.([]interface{})[1].(string)
+			creds, err := containerservice.ListManagedClusterUserCredentials(ctx, &containerservice.ListManagedClusterUserCredentialsArgs{
+				ResourceGroupName: resourceGroupName,
+				ResourceName:      clusterName,
+			})
+			if err != nil {
+				return "", err
+			}
+			encoded := creds.Kubeconfigs[0].Value
+			kubeconfig, err := base64.StdEncoding.DecodeString(encoded)
+			if err != nil {
+				return "", err
+			}
+			return string(kubeconfig), nil
+		}))
+
+    {{% /choosable %}}
+    {{% choosable language "python" %}}
+
+        # Export kubeconfig
+        encoded = creds.kubeconfigs[0].value
+        kubeconfig = encoded.apply(
+            lambda enc: base64.b64decode(enc).decode())
+        pulumi.export("kubeconfig", kubeconfig)
+
+        {{% /choosable %}}
+        {{% choosable language "typescript" %}}
+
+    {{% /choosable %}}
+    {{% choosable language "typescript" %}}
+
+        const creds = pulumi.all([cluster.name, resourceGroup.name]).apply(([clusterName, rgName]) => {
+            return containerservice.listManagedClusterUserCredentials({
+                resourceGroupName: rgName,
+                resourceName: clusterName,
+            });
+        });
+
+        const encoded = creds.kubeconfigs[0].value;
+        export const kubeconfig = encoded.apply(enc => Buffer.from(enc, "base64").toString());
+
+    {{% /choosable %}}
+{{< /chooser >}}
+
+**Step 4**: Connect to the AKS cluster
+
+We can use the `kubeconfig` file connect to the cluster. It's common practice to copy the `kubeconfig` file to `~/.kube/config` which is the directory that `kubectl` looks for the file. However, you can use any directory with the `--kubeconfig` flag, e.g.,
+
+```bash
+$ kubectl --kubeconfig /path/to/kubeconfig_file get pods
+```
+
+As this example shows, deploying an AKS cluster requires setting parameters. This can be done in the portal or with code. The advantage of code is that clusters can be created on demand without having to use the Azure portal. This is convenient when you have different environments, such as a dev/test environment and a production environment.
+
+The complete code for deploying AKS is available on GitHub for [Typescript](https://github.com/pulumi/examples/tree/master/azure-ts-aks), [Python](https://github.com/pulumi/examples/tree/master/azure-py-aks), [Go](https://github.com/pulumi/examples/tree/master/azure-go-aks), and [C#](https://github.com/pulumi/examples/tree/master/azure-cs-aks).
+
+## Summary
+
+Azure Kubernetes Service let's you deploy a Kubernetes cluster quickly and efficiently. An AKS deployment configures the worker nodes since the master nodes are provided by Azure. The Azure Portal provides a simplified interface for configuration, but if you want fine-grain control over the deployment, infrastructure as code is an option. This is particularly true for production deployments where the configuration requires tuning for an application deployment.
```

Comment:
```suggestion
Azure Kubernetes Service lets you deploy a Kubernetes cluster quickly and efficiently. An AKS deployment configures the worker nodes since the master nodes are provided by Azure. The Azure Portal provides a simplified interface for configuration, but if you want fine-grain control over the deployment, infrastructure as code is an option. This is particularly true for production deployments where the configuration requires tuning for an application deployment.
```